### PR TITLE
DeveloperGuide.adoc: fix diagram captions

### PR DESCRIPTION
**Pull request by khooroko**
ignore commit names

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -262,46 +269,53 @@ public class DeleteCommand extends UndoableCommand {
 
 Suppose that the user has just launched the application. The `UndoRedoStack` will be empty at the beginning.
 
-The user executes a new `UndoableCommand`, `delete 5`, to delete the 5th person in the address book. The current state of the address book is saved before the `delete 5` command executes. The `delete 5` command will then be pushed onto the `undoStack` (the current state is saved together with the command).
+The user executes a new `UndoableCommand`, `delete 5`, to delete the 5th person in the address book. The current state of the address book is saved before the `delete 5` command executes. The `delete 5` command will then be pushed onto the `undoStack` (the current state is saved together with the command). This is shown by Figure 3.1.2a.
```
Comment (jelneo) [inline docs/DeveloperGuide.adoc]: shown in* Figure...

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -262,46 +269,53 @@ public class DeleteCommand extends UndoableCommand {
...
 
-The user now decides to execute a new command, `clear`. As before, `clear` will be pushed into the `undoStack`. This time the `redoStack` is no longer empty. It will be purged as it no longer make sense to redo the `add n/David` command (this is the behavior that most modern desktop applications follow).
+The user now decides to execute a new command, `clear`. As before, `clear` will be pushed into the `undoStack`. This time the `redoStack` is no longer empty. It will be purged as it no longer make sense to redo the `add n/David` command. This is shown in Figure 3.1.2e.
 
 image::UndoRedoNewCommand2StackDiagram.png[width="800"]
+_Figure 3.1.2e: Adding a new command when redoStack is not empty_
 
-Commands that are not undoable are not added into the `undoStack`. For example, `list`, which inherits from `Command` rather than `UndoableCommand`, will not be added after execution:
+Commands that are not undoable are not added into the `undoStack`. For example, `list`, which inherits from `Command` rather than `UndoableCommand`, will not be added after execution. This is shown by Figure 3.1.2f.
```
Comment (jelneo) [inline docs/DeveloperGuide.adoc]: shown in* Figure...

**Review by lawwman — APPROVED** [review]
Seems good.

**Review by jaivigneshvenugopal — APPROVED** [review]
nice!

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -169,7 +169,9 @@ image::LogicClassDiagram.png[width="800"]
 _Figure 2.3.1 : Structure of the Logic Component_
 
 image::LogicCommandClassDiagram.png[width="800"]
-_Figure 2.3.2 : Structure of Commands in the Logic Component. This diagram shows finer details concerning `XYZCommand` and `Command` in Figure 2.3.1_
+_Figure 2.3.2 : Structure of Commands in the Logic Component_
+
+Figure 2.3.1 shows the structure of the `Logic` component, while Figure 2.3.2 shows finer  details concerning `XYZCommand` and `Command` in Figure 2.3.1.
```
Comment (jaivigneshvenugopal) [inline docs/DeveloperGuide.adoc]: Should we break this up and chunk it under respective diagrams ? Or this is fine?

**Review by jelneo — APPROVED** [review]

**Pull request merged**